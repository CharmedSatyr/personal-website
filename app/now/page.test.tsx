import { render, screen } from "@testing-library/react";

import Now from "app/now/page";

describe("Now", () => {
	it("renders a heading", () => {
		render(<Now />);

		const heading = screen.getByRole("heading", { level: 1 });

		expect(heading).toBeInTheDocument();
	});

	it("renders four list items", () => {
		render(<Now />);

		const listItem = screen.getAllByRole("listitem");

		expect(listItem.length).toBe(4);
	});

	it("renders four links", () => {
		render(<Now />);

		const links = screen.getAllByRole("link");

		expect(links.length).toBe(4);
	});

	it("should match the last snapshot", () => {
		const { container } = render(<Now />);

		expect(container).toMatchSnapshot();
	});
});
