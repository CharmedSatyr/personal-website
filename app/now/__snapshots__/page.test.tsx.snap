// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Now should match the last snapshot 1`] = `
<div>
  <section
    class="container min-h-96 max-w-3xl"
  >
    <h1>
      Now
    </h1>
    <aside
      class="italic"
    >
      This is my 
      <a
        href="https://nownownow.com/about"
        rel="noopener noreferrer"
        target="_blank"
      >
        Now
      </a>
       page.
    </aside>
    <ol
      class="list-decimal"
    >
      <li>
        In January, my amazing partner Inga
         
        <span
          aria-label="heart"
          role="img"
        >
          ♥️
        </span>
         
        and I got married in an unplanned ceremony in Las Vegas.
      </li>
      <li>
        I am looking for work as a software developer.
         
        <a
          href="/contact"
        >
          Send me a message
        </a>
         if you think I might be right for a project or role you're hiring for.
      </li>
      <li>
        I am writing an adventure module for
         
        <a
          href="https://dnd.wizards.com/how-to-play"
          rel="noopener noreferrer"
          target="_blank"
        >
          Dungeons & Dragons 5e
        </a>
        .
      </li>
      <li>
        I have an idea for a digital game and some early prototypes built in
         
        <a
          href="https://learn.unity.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          Unity
        </a>
        .
      </li>
    </ol>
  </section>
</div>
`;
